### ISSUE TITLE
Add vertex model deployment resource

### ISSUE THREAD

**Issue by alwhelan22** [hashicorp/terraform-provider-google#15303]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

I'd like to be able to deploy a vertex ai model from model registry (not a GCP bucket) to a designated/existing endpoint.

### New or Affected Resource(s)

* google_ml_engine_model

### Potential Terraform Configuration

```tf
resource "google_ml_engine_model" "default" {
  name        = "default"
  description = "My model"
  regions     = ["us-central1"]
  endpoint = google_vertex_ai_endpoint.endpoint.id
  labels = {
    my_model = "foo"
  }
  online_prediction_logging         = true
  online_prediction_console_logging = true
}

resource "google_vertex_ai_endpoint" "endpoint" {
  name         = "endpoint-name"
  display_name = "sample-endpoint"
  description  = "A sample vertex endpoint"
  location     = "us-central1"
}
```

### References

* https://github.com/GoogleCloudPlatform/magic-modules/pull/6661

**Comment by EduardJoy** [comment]
Greetings, any update on this issue particularly?

**Comment by EduardJoy** [comment]
@mattjudge 
I think this was already implemented.
it should be  google_vertex_ai_index_endpoint_deployed_index

**Comment by mattjudge** [comment]
I don't think so. My understanding is that this issue is looking for a terraform implementation of https://cloud.google.com/vertex-ai/docs/predictions/deploy-model-console#deploy_a_model_from_the_page, whereas  `google_vertex_ai_index_endpoint_deployed_index` looks like it's implementing https://cloud.google.com/vertex-ai/docs/vector-search/deploy-index-public

The [docs](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vertex_ai_endpoint#deployed_models-1)  for the terraform provider's `vertex_ai_endpoint` state
> [deployed_models](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/vertex_ai_endpoint#deployed_models-1) - Output only. The models deployed in this Endpoint. To add or remove DeployedModels use EndpointService.DeployModel and EndpointService.UndeployModel respectively.

but I'm unaware of any way to actually do this (that string looks like it comes from the Java documentation, where that is actually possible).

**Comment by mattjudge** [comment]
It looks like this PR is the work in progress https://github.com/GoogleCloudPlatform/magic-modules/pull/12490

**Comment by cperalta-corsha** [comment]
Any updates here? It looks like the associated PR was closed and never merged. Is this work being tracked somewhere else?
